README.md

```diff
@@ -62,26 +62,29 @@ linear runtime. Can we do better?
 
 ## Solution 2
 
-Our goal with this improved solution is to improve the time complexity of the
+Our goal with this second solution is to improve the time complexity of the
 `#increment` method. That means that we need to find a way to increment each
 value in the stack _without_ iterating over every element in the array when the
 `#increment` method is called.
 
 One very useful piece of information is that we only need to check the values of
 the array _when the `#pop` method is called_. So we don't need to increment each
 value in the `data` array when the `#increment` method is called, we just need a
-way to _keep track of all the `#increment` operations_, so that we can apply the
+way to _keep track of all the `#increment` operations_. We can then apply the
 correct incremented value when the `#pop` method is called.
 
-To do this, we'll make a second array `inc` that has the same number of elements
-as the `data` array, and this array will be used to keep track of all the
-`#increment` operations. The `inc` array will keep track of the number of
-elements that need to be incremented by storing that information at an index
-position that corresponds to the number of elements, and will encode the data
-about how many times the corresponding element from the `data` array needs to be
-incremented when the `#pop` method is called.
+To do this, we'll make a second array `inc` that will have the same number of
+elements as the `data` array. The `inc` array will keep track of the **number of
+elements that need to be incremented** by storing a value at the index position
+that corresponds to that number; the **value** stored will indicate the number
+of times those elements should be incremented. When the `#pop` method is called,
+the element will be incremented as indicated by the corresponding element in the
+`inc` array; in addition, if there are any remaining elements in the stack, the
+increment information will be passed down to the next element in the `inc` array
+so that those remaining elements also get incremented when they are popped off
+the stack.
 
-That is probably easier to visualize with some diagrams:
+The process is probably easier to visualize with some diagrams:
 
 ![Stack Plus Solution](https://curriculum-content.s3.amazonaws.com/phase-5/phase-5-practice-challenge-stack-plus/stack-plus-solution.png)
 
```

Comment:
I also have a couple recommendations for edits to the diagram: 

- Below the 'push(5)' box, change the text to "Because 5 was pushed after increment was called, it does not need to be incremented when it is popped off the stack"
- Next to the first 'pop' box on the second row, change the beginning of the second sentence to: "Remember, because we called... "

Reply:
Done!